### PR TITLE
[FLINK-25340][testutils] Remove JM temp files before stopping container to avoid file permission issues

### PR DESCRIPTION
**Pull request by PatrickRen**
## What is the purpose of the change

This pul request fixes file permission issues when deleting temporary directories and files when stopping `FlinkContainers`. Instead JobManager container will cleanup temporary checkpoint and HA recovery files before stopping.

## Brief change log

- Remove JM temp files before stopping container
- Re-enable temporary disabled `KafkaSourceE2ECase`

## Verifying this change

This change is already covered by existing `KafkaSourceE2ECase`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkContainers.java [hunk 1]
```diff
@@ -160,6 +162,7 @@ public void stop() {
             restClusterClient.close();
         }
         this.taskManagers.forEach(GenericContainer::stop);
+        deleteJobManagerTemporaryFiles();
```
Comment (fapaul) [inline flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkContainers.java:165]: Do we still need the cleanup hook passed to this class? AFAICT the jobmanager deletes the HA and the checkpoint data so only the image has to be deleted. Can we delete the image immediately after we have built the containers in the builder already?

File: flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkContainers.java [hunk 2]
```diff
@@ -317,4 +320,25 @@ private void waitUntilAllTaskManagerConnected() throws InterruptedException, Tim
...
+                            Paths.get(conf.get(HighAvailabilityOptions.HA_STORAGE_PATH), "*")
+                                    .toString());
+            if (result.getExitCode() != 0) {
+                throw new IllegalStateException(
+                        String.format(
+                                "rm command returned non-zero exit code %d. \nSTDOUT: %s\nSTDERR: %s",
+                                result.getExitCode(), result.getStdout(), result.getStderr()));
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to delete temporary files in JobManager");
```
Comment (fapaul) [inline flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkContainers.java:341]: Nit: Maybe add the directory names to the exception.

Comment (fapaul) [reply flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkContainers.java:341]: We should not swallow the causing and exception and pass it to the RuntimeException

**Review by fapaul — COMMENTED** [review]
I have some smaller questions but overall looks good let's wait for the CI.

**Comment by fapaul** [comment]
The second commit should probably have this ticket number https://issues.apache.org/jira/browse/FLINK-25263

File: flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkContainers.java (hunk 1 above)
Comment (PatrickRen) [reply flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkContainers.java:165]: I prefer to keep it for deleting temporary directories that are created in `FlinkContainersBuilder`. JM only removes dirs and files it created inside the temp dir (`/flink/checkpoint/{jobid}/xxx`) but not the temp dir itself (`/flink/checkpoint`).

Currently the image will be deleted when JVM exits. The image will be kept only if you use `new ImageFromDockerfile(name, false)`, where the second parameter is `deleteOnExit`.
Code for building Flink image: https://github.com/apache/flink/blob/master/flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkImageBuilder.java#L215

**Comment by PatrickRen** [comment]
Thanks for the review @fapaul ! I made another push just now. Please take a look when you are available

**Review by fapaul — APPROVED** [review]
Thanks for fixing this 👍

**Pull request merged**